### PR TITLE
X509 federation

### PR DESCRIPTION
**Pull request by EZIOJQ**
This PR adds support for X.509 federation support on workload identity pool. It fixes https://github.com/hashicorp/terraform-provider-google/issues/18812. 

For more details about how this feature works, check out the public doc here https://cloud.google.com/iam/docs/workload-identity-federation-with-x509-certificates

```release-note:enhancement
iambeta: added `x509` field to `google_iambeta_workload_identity_provider` resource
```

### CODE REVIEW

**Comment by EZIOJQ** [comment]
I don't know how to re-trigger the VCR test, I fixed some visibility issues and manual test run has succeeded

**Review by NickElliot — CHANGES_REQUESTED** [review]
mostly LGTM -- just want to get one thing changed around. could you add the pem_certificates as files in `mmv1/third_party/terraform/services/iambeta/test-fixtures`?

can see examples [here](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/services/compute/test-fixtures)

thanks!

File: mmv1/templates/terraform/examples/iam_workload_identity_pool_provider_x509_basic.tf.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
...NVBAMMGkdvb2dsZSAqKlRlc3RpbmcqKiBSb290IENBMCAXDTIwMDEwMTAw\nMDAwMFoYDzIxMzAwMTAxMDAwMDAwWjBcMRcwFQYDVQQKEw5Hb29nbGUgVEVTVElO\nRzEcMBoGA1UECxMTR29vZ2xlIFRFU1RJTkcgdW5pdDEjMCEGA1UEAwwaR29vZ2xl\nICoqVGVzdGluZyoqIFJvb3QgQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK\nAoIBAQDHqdf17AA4iwl+HGtTD/qJJv9XZTjzjHiaRYWuGQ3iddf1LhYklLqHvpqO\nCBlhkg6NIZmoDPKicEi8pfGSzp6btcElyrr1ekECk5jEBcdl6tX/gTSYv7v1h9Dk\nSJDHBAnoSsVW0/PNwI+YGEE2kizGMeg1moXlTHEB+yeGCik/+4eVRas/+wrlrTE5\nlMFMq8WhdnBx6udcc/BEauvlTybHaN3rJUVpVrJWeVoPGGtXR6MJrdbVScn9GYOL\nP6sXMPTr+pTY6ebrjs7K/wTVQqLJ1zArCvAEH8FvuAhiI19yM1uBRAds/fJbSUvF\nsEiIlDC4y7DAK7yWHUAjQIBV1xhDAgMBAAGjejB4MA4GA1UdDwEB/wQEAwICBDAd\nBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDwYDVR0TAQH/BAUwAwEB/zAZ\nBgNVHQ4EEgQQxEHfoOjPZleWgcpHJnyE3TAbBgNVHSMEFDASgBDEQd+g6M9mV5aB\nykcmfITdMA0GCSqGSIb3DQEBCwUAA4IBAQBcECwxgaK4ZgYO7guayK5QRpTb3Y6z\npkmSOkj9h1+HF5Ch/o5FiweJi8k28h9Mfz//gKU2cXXWfXuY81CFEBstjw7jYt3+\nd4owS5sYKu5WswGj4jsQXRrt0tO0+0UngP560eggFCyLChG75lp54r92hTPk4fY4\nvarURZH7X0jg9W7MO6E6/HmKdMIxanuWdkbpPe6kj5I7SNvVhqsncoga8iZJ6QK/\nrsC2LTax4dxUUSkP5vmwwiYbgXYbk9JcXI+OyVcMvVxN/akI9/JgYHOol6NdChTB\nwU0yjNb9B5TrgtP1/fs+LugINrN1R1hgVHDVlE4mwHej+5XL6m9xAxkc\n-----END CERTIFICATE-----"
```
[the remark this answers is not in the document]
Comment (EZIOJQ) [reply mmv1/templates/terraform/examples/iam_workload_identity_pool_provider_x509_basic.tf.erb]: Done in the new update

**Review by NickElliot — CHANGES_REQUESTED** [review]
> Error: Error creating WorkloadIdentityPoolProvider: googleapi: Error 400: Certificate is not in PEM format. Please make sure to include the suffix, prefix and the new line characters.

make sure to format the files correctly!

**Comment by EZIOJQ** [comment]
@NickElliot I've changed the file format, PTAL, thanks!

**Comment by EZIOJQ** [comment]
I closed this as my repo was messed up with unrelated changes. Created a new pull request here https://github.com/GoogleCloudPlatform/magic-modules/pull/11493

**Pull request closed without merging**